src/network/serverpackethandler.cpp

```diff
@@ -630,30 +649,24 @@ void Server::handleCommand_InventoryAction(NetworkPacket* pkt)
 		if (ma->from_inv != ma->to_inv)
 			m_inventory_mgr->setInventoryModified(ma->to_inv);
 
-		bool from_inv_is_current_player = false;
-		if (ma->from_inv.type == InventoryLocation::PLAYER) {
-			if (ma->from_inv.name != player->getName())
-				return;
-			from_inv_is_current_player = true;
-		}
-		
-		bool to_inv_is_current_player = false;
-		if (ma->to_inv.type == InventoryLocation::PLAYER) {
-			if (ma->to_inv.name != player->getName())
-				return;
-			to_inv_is_current_player = true;
+		if (!check_inv_access(ma->from_inv) ||
+				!check_inv_access(ma->to_inv)) {
+			delete a;
```

Comment:
grrrr - unique_ptr would be preferred for this reason

Reply:
That for sure, but it's not the goal of this PR to fix this pre-C++11 code.

Reply:
I think fixing code like this should be done incrementally, to make it easier to tackle. Like how we fix code style